src/components/PullRequestList/index.tsx

```diff
@@ -0,0 +1,110 @@
+import { FC, useState, useEffect } from 'react';
+import Head from '@/components/head';
+import Layout from '@/components/Layout';
+import PullRequest from '@/components/pullRequests';
+import CardShimmer from '@/components/Loaders/cardShimmer';
+import axios from 'axios';
+import styles from './PullRequestList.module.scss';
+
+type pullRequestType = {
+  title: string;
+  username: string;
+  createdAt: string;
+  updatedAt: string;
+  url: string;
+};
+
+type PullRequestListProps = {
+    prType: string;
+};
+
+const PullRequestList: FC<PullRequestListProps> = ({ prType }) => {
+  const [pullRequests, setPullRequests] = useState<pullRequestType[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState('');
+  const [page, setPage] = useState(1);
+
+  const prUrl = `${process.env.NEXT_PUBLIC_BASE_URL}/pullrequests/${prType}?page=${page}`;
+
+  const fetchPRs = async () => {
+    try {
+      setLoading(true);
+      const response = await axios.get(prUrl);
+      setPullRequests(response.data.pullRequests);
+
+      if (!response.data.pullRequests.length) {
```

Comment:
NIT: For readability purpose, you can destructure the `response.data` like this
```javascript
const { pullRequests } = response.data
```
And then use this `pullRequests` across.
Will make it more readable

Reply:
I wanted to just set it and use the state of pullRequests instead
but somehow the state was not changing synchronously in my machine 😓 

Otherwise the if condition will check on the state pullRequests instead of the property buried inside the response.
Please test if it works on your machine.